video/out/wayland_common.c

```diff
@@ -2683,6 +2683,23 @@ static int handle_round(int scale, int n)
     return (scale * n + WAYLAND_SCALE_FACTOR / 2) / WAYLAND_SCALE_FACTOR;
 }
 
+static bool hdr_metadata_valid(struct pl_hdr_metadata *hdr)
+{
+    // Always return a hard failure if this condition fails.
+    if (hdr->min_luma >= hdr->max_luma)
+        return false;
+
+    // If max_cll or max_fall are invalid, set them to zero.
+    if ((hdr->max_cll && (hdr->max_cll <= hdr->min_luma || hdr->max_cll > hdr->max_luma)) ||
+        (hdr->max_fall && (hdr->max_fall <= hdr->min_luma || hdr->max_fall > hdr->max_luma)) ||
+        (hdr->max_cll && hdr->max_fall && hdr->max_fall > hdr->max_cll)) {
+        hdr->max_cll = 0;
+        hdr->max_fall = 0;
+    }
```

Comment:
I think it's better to treat them separately.

Reply:
Yes. You can have either of them, both or none.

Reply:
`max_cll` and `max_fall` are optional, but when they are defined, they need to follow Wayland protocol specific restriction. See how mesa checks them https://gitlab.freedesktop.org/mesa/mesa/-/blob/cd9c9a83d64d3b4231c45cc7b391332efeaaeaec/src/vulkan/wsi/wsi_common_wayland.c#L1169-1187

Reply:
Right but mesa doesn't set any metadata at all if any part of the Wayland restriction fails.

Reply:
I don't see why it matters.